app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java

```diff
@@ -43,24 +41,23 @@
 import fr.free.nrw.commons.kvstore.JsonKvStore;
 import fr.free.nrw.commons.location.LatLng;
 import fr.free.nrw.commons.nearby.Place;
+import fr.free.nrw.commons.upload.UploadMediaDetail;
+import fr.free.nrw.commons.upload.UploadMediaDetailAdapter;
 import fr.free.nrw.commons.settings.Prefs;
-import fr.free.nrw.commons.upload.Description;
-import fr.free.nrw.commons.upload.DescriptionsAdapter;
 import fr.free.nrw.commons.upload.SimilarImageDialogFragment;
-import fr.free.nrw.commons.upload.Title;
 import fr.free.nrw.commons.upload.UploadBaseFragment;
 import fr.free.nrw.commons.upload.UploadModel;
 import fr.free.nrw.commons.upload.UploadModel.UploadItem;
 import fr.free.nrw.commons.utils.DialogUtil;
 import fr.free.nrw.commons.utils.ImageUtils;
 import fr.free.nrw.commons.utils.ViewUtil;
-import io.reactivex.disposables.Disposable;
+
 import timber.log.Timber;
 
 import static fr.free.nrw.commons.utils.ImageUtils.getErrorMessageForResult;
 
 public class UploadMediaDetailFragment extends UploadBaseFragment implements
-        UploadMediaDetailsContract.View {
+        UploadMediaDetailsContract.View, UploadMediaDetailAdapter.EventListener {
```

Comment:
Instead of implementing the interface, should we not use anonymous inner classes ?

Reply:
I didn't get this. I am implementing this interface to trigger calls from the Adapter to the fragment. Which is the other class that can help with it?